src/spikeinterface/core/core_tools.py

```diff
@@ -934,3 +941,72 @@ def convert_bytes_to_str(byte_value: int) -> str:
         byte_value /= 1024
         i += 1
     return f"{byte_value:.2f} {suffixes[i]}"
+
+
+def spike_vector_to_dict(spike_vector: np.ndarray) -> list[dict]:
```

Comment:
`dict` is a very unspecific name here.

Can you just use `spike_vector_to_spike_trains` or `spike_trains_dict` at the end if you want to be more specific. 

Then in the documentation you can write that the dicts are dictionary where spike_train_dict[unit_index] gives you all the spikes frames for a given unit. 

I know the library to some degree and it took me a while to understand what you are doing.

Reply:
Actually I really agree with this. In the library sometimes I forget what is a dict and what is a np.array and so it is hard to follow indexing efforts in general.

Reply:
Done :+1: